projects/observability/src/shared/dashboard/data/graphql/explore/explore-cartesian-data-source.model.ts

```diff
@@ -41,25 +42,33 @@ export abstract class ExploreCartesianDataSourceModel extends GraphQlDataSourceM
   protected fetchResults(interval: TimeDuration | 'AUTO'): Observable<CartesianResult<ExplorerData>> {
     const requestState = this.buildRequestState(interval);
     const timeRange = this.getTimeRangeOrThrow();
+
     if (requestState === undefined) {
       return NEVER;
     }
 
     return this.query<ExploreGraphQlQueryHandlerService>(inheritedFilters =>
       this.buildExploreRequest(requestState, this.getFilters(inheritedFilters), timeRange)
     ).pipe(
-      mergeMap(response =>
-        this.mapResponseData(requestState, response, requestState.interval as TimeDuration, timeRange)
+      mergeMap(
+        response =>
+          this.mapResponseData(requestState, response, requestState.interval as TimeDuration, timeRange) as Observable<
+            CartesianResult<ExplorerData>
+          >
       )
     );
   }
 
   protected mapResponseData(
     requestState: ExploreRequestState,
     response: GraphQlExploreResponse,
-    interval: TimeDuration,
+    interval: TimeDuration | undefined,
     timeRange: GraphQlTimeRange
-  ): Observable<CartesianResult<ExplorerData>> {
+  ): Observable<CartesianResult<ExplorerData>> | Observable<GraphQlExploreResult> {
+    if (interval === undefined && requestState.groupBy === undefined) {
+      return of(response.results[0]);
```

Comment:
If you don't mind, can we add a check that `reponse.results.length > 0` here before returning the `0`th index?

Reply:
✅ added